contracts/BatchTransfer.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.2 <0.8.0;
+
+import "protocol-contracts/tokens/contracts/erc-1155/ERC1155Base.sol";
+
+contract BatchTransfer {
+    /// @notice Tokens on the given ERC-1155 contract are transferred from you to the recipients.
+    ///         Don't forget to execute setApprovalForAll first to authorize this contract.
+    function batchTransfer(address contractAddress, address[] memory recipients, uint256[] memory tokenIds, uint256[] memory amounts) external {
+        require(recipients.length > 0, "Recipients are not defined");
+        require(
+            recipients.length == tokenIds.length && tokenIds.length == amounts.length,
+            "Invalid array data"
+        );
+
+
+        for (uint256 index; index < recipients.length; index++) {
+            ERC1155Base(contractAddress).safeTransferFrom(msg.sender, recipients[index], tokenIds[index], amounts[index], "");
```

Comment:
Would it make sense to check the balance before the transfer?
So another `require` statement?

Get balance -> require to be more or equal to the amount

Reply:
yes, good point. I will add this

Reply:
Updated

Reply:
Now there is another point of view 😄 
Do we prevent all the transfers just because one of the NFTs has insufficient balance?
If we do that we should at least add the tokenId to the error message 😄

Reply:
Reverted balance checking changes